File: Data Structures & Algorithms/kth-largest-element-in-an-array/submission-0.py
class Solution:
    def findKthLargest(self, nums: List[int], k: int) -> int:
        #run quickselect but 
        #run fastselect most optimal 
        def partition(arr,l,r):
            pivot=arr[r]
            i=l
            for j in range(l,r):
                if arr[j]<=pivot:
                    arr[j],arr[i]=arr[i],arr[j]#makes sure before pivot
                    i+=1
            #at end must flip pivot with r 
            arr[r],arr[i]=arr[i],arr[r]
            return i#we retunr the index of the pivot
        def quickselect(arr,l,r,k):
            index=partition(arr,l,r)
            if l>=r:
                return index
            if index-l+1==k:
                return index
            if index-l+1>k:
                return quickselect(arr,l,index-1,k)
            #else
            return quickselect(arr,index+1,r,k-(index-l+1))

        #kth largest elmeent in teh array is len(nums)-k

        return nums[quickselect(nums,0,len(nums)-1,len(nums)-k+1)] if nums else None

File: Data Structures & Algorithms/duplicate-integer/submission-0.java
class Solution {
    public boolean hasDuplicate(int[] nums) {
int y = 0;
 for (int i=0; i<nums.length; i++){
    y =0;
    for (int j =0; j<nums.length; j++) {
        if (nums[i] == nums[j]) {
            y++;
        }
    }
    if (y>1){
        return true;
    }
 }
 return false;
    }
}
